File: problems/0271-encode-and-decode-strings/solution.cpp
class Codec {
public:
    char unit_separator = '\x1F';

    // Encodes a list of strings to a single string.
    string encode(vector<string>& strs) {
        string encoded;
        for (string s : strs) {
            encoded.append(s);
            encoded.push_back(unit_separator);
        }
        return encoded;
    }

    // Decodes a single string to a list of strings.
    vector<string> decode(string s) {
        vector<string> solution;
        string str;
        for (char c : s) {
            if (c == unit_separator) {
                solution.push_back(str);
                str = "";
            } else {
                str.push_back(c);
            }
        }
        return solution;
    }
};

// Your Codec object will be instantiated and called as such:
// Codec codec;
// codec.decode(codec.encode(strs));

File: problems/0271-encode-and-decode-strings/solution.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        encoded = ""

        for s in strs:
            encoded += str(len(s)) + '#' + s

        return encoded        

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        ans = []
        i = 0
        while i < len(s):
            l = ""
            while s[i] != '#':
                l += s[i]
                print(l)
                i += 1
            i += 1

            word = s[i : i + int(l)]
            ans.append(word)
            i = i + int(l)

        return ans



# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))
